### ISSUE TITLE
Exchange revaluation difference in showing up in AR

### ISSUE THREAD

**Issue by Vigneshsekar** [frappe/erpnext#43327]
### Information about bug

When we pass Exchange Rate Re-Valuation Entry for a Customer, the entire receivable amount changes as per the revised exchange rate. We do not have invoice-wise reference in it.

And so, at the time of passing payment entry, the system still takes the older exchange rate at which the invoice was generated and also an Exchange gain or loss is passed.

This leads to two issues.

1) Even after receiving full payment in USD from the customer, there is an outstanding in INR. This outstanding corresponds to the difference caused by the exchange revaluation. The user will not be able to reconcile this.

2) The exchange gain or loss is booked twice. One during the revaluation and another during the payment entry as the payment entry reconciles the invoice with the invoice's exchange rate and books the remaining in exchange gain or loss.

### Module

accounts

### Version

ERPNext version - Version 15

### Installation method

FrappeCloud

**Comment by finbyz** [comment]
Dear @ruthra-kumar,

There is no update on this issue even after 2 months, I can think through below solutions.

Option 1: Easy solution for now can be provide functionality in Exchang rate revaluation entry to pass reversal entry, and both should carry Exchange rate revaluation entry reference. This is exactly how we want them to use it as of now, but reversal of revaluation JV is actually not mentioned anywhere as per regulation.

Option 2: Second more permanent solution will be to check at payment entry time if there is any exchange rate revaluation entry against particular invoice, it should affect INR value of that invoice in the payment entry to pass difference only as exchange gain and loss. This is more permanent solution but it need further brainstorming on:
    - What will be the effect of past dated transactions of revaluation.
    - How to handle this in payment reconciliation.
    
If you agree on any we can collaborate to complete this faster.

**Comment by Vigneshsekar** [comment]
I will assign it to myself. 

Will workout both of your options and get back @finbyz .

**Comment by finbyz** [comment]
As discussed on the call

Option 1: As in earlier versions the JV was not reconciled against Exchange revolution entry. So exchange for revaluation entry we could reconcile against it's reversal entry.  Now either we can remove the against entry reference in revaluation entry or pass the same reference in the reversal of Exchange revaluation entry. This will make it work as it was working till date in ERPNext.
Option 2: We discussed other use cases should mostly work, only 1 thing require more testing:
 If transaction sequence is:
1) Invoice created at $ 10000 with exchange rate 80
2) Payment received for partial amount lets say 50%, at rate of 82, gain loss is marked on payment entry with difference of 2.
3) Remaing amount got revalued at exchange rate 84.
4) Remaing payment received at exchange rate 83, and exchange gain loss now passed last 84 for this amount.
5) Now if we cancel first payment entry, how should we capture correct difference in exchange gain and loss, should we just go by date of exchange rate revaluation, or should we go by for effective exchange rate based on all related entries? 
Orignal invoice + Revaluation against it + payment entry + exchange gain loss of that payment entry) / 
foreign currency value of all these payment entries   = Exchange effective exchange rate. 

Let me know if you need to brainstorm the solution deeper.

**Comment by Vigneshsekar** [comment]
Adding more context to the same issue mentioned above. 

> The exchange gain or loss is booked twice. One during the revaluation and another during the payment entry as the payment entry reconciles the invoice with the invoice's exchange rate and books the remaining in exchange gain or loss.

Example case:

**01-04-2023** | Journal Entry | ACC-JV-2025-00001 | JPY 105,375,000 @ 0.6184 = INR 6,51,63,900

**31-03-2024** | Exchange Rate Revaluation | ACC-JV-2025-00002 | JPY 105,375,000 @ 0.55148 = INR 5,81,12,205 | Gain/Loss = -70,51,695

**31-03-2025** | Exchange Rate Revaluation | ACC-JV-2025-00003 | JPY 105,375,000 @ 0.57178 = INR 6,02,51,317.50 | Gain/Loss = 21,39,112.50

Now, doing the payment and reconciling it with the source document made on 01-04-2023.

**07-05-2025** | Journal Entry | ACC-JV-2025-00004 | JPY 105,375,000 @ 0.5865 = INR 6,18,02,437.50

**07-05-2025** | Gain/Loss Voucher | ACC-JV-2025-00005 | System posted loss = -33,61,462.50

The expected Gain/Loss should be calculated as:

**Paid Amount (6,18,02,437.50) – Last Revalued Balance (6,02,51,317.50) = 15,51,120**

But the system auto-generated voucher shows **-33,61,462.50**, which is incorrect.

The multiple bookings made to Exchange Gain/Loss should be fixed asap.
